File: Stack Permutations.java
class Solution {
    public static int isStackPermutation(int n, int[] ip, int[] op) {
        // code here
        Stack<Integer> s=new Stack<>();
        int i=0;
        int j=0;
        while(j<n){
            while(i<n && (s.isEmpty() || s.peek()!=op[j])){
                s.push(ip[i]);
                i++;
            }
            while(!s.isEmpty() && s.peek()==op[j]){
                s.pop();
                j++;
            }
            if(i==n && !s.isEmpty() && s.peek()!=op[j])return 0;
        }
        return 1;
    }
}

File: Stack Permutations.py
def isStackPermutation(N, A, B):
        arr = []
        i = j = 0
        while True:
            if arr and arr[-1] == B[j]:
                if j>=len(B): break
                arr.pop()
                j+=1
            else:
                if i>=len(A): break
                arr.append(A[i])
                i+=1
        return 1 if len(arr) == 0 else 0
